client/finality-grandpa/src/communication/mod.rs

```diff
@@ -290,22 +291,20 @@ impl<B: BlockT, N: Network<B>> NetworkBridge<B, N> {
 		});
 
 		let topic = round_topic::<B>(round.0, set_id.0);
-		let incoming = Compat::new(self.gossip_engine.messages_for(topic)
-			.map(|item| Ok::<_, ()>(item)))
-			.filter_map(|notification| {
+		let incoming = self.gossip_engine.messages_for(topic)
+			.filter_map(move |notification| {
 				let decoded = GossipMessage::<B>::decode(&mut &notification.message[..]);
-				if let Err(ref e) = decoded {
-					debug!(target: "afg", "Skipping malformed message {:?}: {}", notification, e);
-				}
-				decoded.ok()
-			})
-			.and_then(move |msg| {
-				match msg {
-					GossipMessage::Vote(msg) => {
+
+				match decoded {
+					Err(ref e) => {
+						debug!(target: "afg", "Skipping malformed message {:?}: {}", notification, e);
+						return ready(None);
```

Comment:
I'd prefer if this read `future::ready` instead. Wasn't immediately obvious to me.

Reply:
I agree. As a rule of thumb, importing free functions into the local namespace hurts readability.

`module::function` lets us know where to look for the function without searching the imports.
`function` lets us know that the function is in the current module.

Reply:
Good point. Fixed with https://github.com/paritytech/substrate/pull/4691/commits/6858319a169babd168871414883bbb969cc8c32d.